ui/components/component-library/button-primary/button-primary.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import { ButtonBase } from '../button-base';
+import { BUTTON_SIZES } from './button-primary.constants';
+
+export const ButtonPrimary = ({ className, children, danger, ...props }) => {
```

Comment:
Not sure if it makes sense to repeat some of the base button props here or not, maybe something we can user test to see if it would make engineers lives easier or not. Fine with extending ButtonBase props for now.

Reply:
Right. It reduces duplicate code but creates more cognitive load for the engineer to now go into ButtonBase and see what props are available for ButtonPrimary instead of having them visible in this component. Might not be a big deal. But I'll add it to our user test table

Reply:
Oh sorry I meant is it worth **including** all of the props from ButtonBase like block, etc not reduce them.